File: code2map/docs/examples/java/output/parts/UserManagementService_validateUserName.java
// code2map fragment (non-buildable)
// original: docs/examples/java/UserManagementService.java
// lines: 178-186
// symbol: UserManagementService#validateUserName
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls isEmpty, userName.length, userName.trim
    private void validateUserName(String userName) {
        if (userName == null || userName.trim().isEmpty()) {
            throw new IllegalArgumentException("ユーザー名は必須です");
        }
        if (userName.length() > MAX_USER_NAME_LENGTH) {
            throw new IllegalArgumentException(
                "ユーザー名は" + MAX_USER_NAME_LENGTH + "文字以内で入力してください");
        }
    }

File: code2map/docs/examples/python/output/parts/UserManagementService__validate_user_name.py
# code2map fragment (non-buildable)
# id: CD15
# original: docs/examples/python/user_management_service.py
# lines: 211-215
# symbol: UserManagementService#_validate_user_name
# notes: references dataclasses.dataclass, re, typing.Optional; calls ValueError, len, user_name.strip
    def _validate_user_name(self, user_name: str) -> None:
        if not user_name or not user_name.strip():
            raise ValueError("ユーザー名は必須です")
        if len(user_name) > self.MAX_USER_NAME_LENGTH:
            raise ValueError(f"ユーザー名は{self.MAX_USER_NAME_LENGTH}文字以内で入力してください")
